### PR TITLE
ddl: check the tiflash replica count when setting tiflash replica 

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve?

Fix https://github.com/pingcap/tidb/issues/18810

When setting the tiflash replica count, TiDB should check the replica count not more than the count of tiflash store, otherwise, the tiflash region replica will never available. 

example:

```sql
test> alter table t set tiflash replica 2 location labels 'a','b';
(1105, 'the tiflash replica count: 2 should less than the total tiflash server count: 1')
test> alter table t set tiflash replica 1 location labels 'a','b';
Query OK, 0 rows affected
```

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM

### Release note 

- Check the tiflash replica count when setting tiflash replica

### CODE REVIEW

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -3905,6 +3910,18 @@ func (d *ddl) AlterTableSetTiFlashReplica(ctx sessionctx.Context, ident ast.Iden
 	return errors.Trace(err)
 }
 
+func checkTiFlashReplicaCount(ctx sessionctx.Context, replicaCount uint64) error {
+	// Check the tiflash replica count should less than the total tiflash store.
```
Comment (AilinKid) [inline ddl/ddl_api.go:3914]:
```suggestion
	// Check the tiflash replica count should be less than that of the total tiflash stores.
```

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -3905,6 +3910,18 @@ func (d *ddl) AlterTableSetTiFlashReplica(ctx sessionctx.Context, ident ast.Iden
...
+	tiflashStoreCnt, err := infoschema.GetTiFlashStoreCount(ctx)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if replicaCount > tiflashStoreCnt {
+		return errors.Errorf("the tiflash replica count: %d should less than the total tiflash server count: %d", replicaCount, tiflashStoreCnt)
```
Comment (AilinKid) [inline ddl/ddl_api.go:3920]: ditto

File: infoschema/tables.go [hunk 3]
```diff
@@ -1457,6 +1459,27 @@ func GetStoreServerInfo(ctx sessionctx.Context) ([]ServerInfo, error) {
 	return servers, nil
 }
 
+// GetTiFlashStoreCount returns the count of tiflash server.
+func GetTiFlashStoreCount(ctx sessionctx.Context) (uint64, error) {
```
Comment (AilinKid) [inline infoschema/tables.go:1463]:
```suggestion
func GetTiFlashStoreCount(ctx sessionctx.Context) (cnt uint64, err error) {
```
and remove the declaration below

**Comment by JaySon-Huang** [comment]
Maybe we should remove TiFlash store(s) whose status is "tombstone" from the valid store count? @crazycs520  @zanmato1984

**Comment by crazycs520** [comment]
@JaySon-Huang, the `tombstone` tiflash store is already been remove before the check.

**Pull request merged**